### PR TITLE
*: Fix newly raised clippy warnings

### PR DESCRIPTION
**Pull request by umgefahren**
# Description

I corrected warnings raised by clippy that accumulated or came from a new iteration of clippy.

## Change checklist

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works

## Commit message body
Fixed minor issues raised by clippy to improve correctness and readablitity.

### CODE REVIEW

**Review by umgefahren — COMMENTED** [review]
Looks good to me.

File: core/CHANGELOG.md [hunk 1]
```diff
@@ -4,8 +4,11 @@
 
 - Hide `prost::Error` from public API in `FromEnvelopeError::InvalidPeerRecord` and `signed_envelope::DecodingError`. See [PR 3058].
 
+- Fixed minor clippy issue. See [PR 3106].
```
Comment (jxs) [inline core/CHANGELOG.md]: I don't think we need to announce clippy improvements in the `CHANGELOG.md` if they don't affect the public API, see [here](https://github.com/libp2p/rust-libp2p/pull/3081/files)

File: examples/chat-tokio.rs [hunk 2]
```diff
@@ -53,7 +53,7 @@ async fn main() -> Result<(), Box<dyn Error>> {
     // Create a random PeerId
     let id_keys = identity::Keypair::generate_ed25519();
     let peer_id = PeerId::from(id_keys.public());
-    println!("Local peer id: {:?}", peer_id);
```
Comment (jxs) [inline examples/chat-tokio.rs:56]: main `cargo.toml` [`rust-version` is `1.60`](https://github.com/libp2p/rust-libp2p/blob/master/Cargo.toml#L4) which was done on https://github.com/libp2p/rust-libp2p/pull/2646 while all other crates are still with `1.56`. Should we also upgrade the other crates for consistency?  @mxinden @thomaseizinger 
In this case captured identifiers were only [introduced in `1.58`](https://blog.rust-lang.org/2022/01/13/Rust-1.58.0.html#captured-identifiers-in-format-strings), I wonder if clippy takes attention to that and there are more cases like this not being linted.

**Review by jxs — APPROVED** [review]
Hi Hannes LGTM! Curious, how did you run `cargo clippy`, did you run the [`custom-clippy`](https://github.com/libp2p/rust-libp2p/blob/master/.cargo/config.toml#L3)?

**Comment by thomaseizinger** [comment]
Related: https://github.com/libp2p/rust-libp2p/pull/3090

File: examples/chat-tokio.rs (hunk 2 above)
Comment (jxs) [reply examples/chat-tokio.rs:56]: right, just noticed https://github.com/libp2p/rust-libp2p/pull/3090#pullrequestreview-1170198498, thanks Thomas!

Comment (umgefahren) [reply examples/chat-tokio.rs:56]: Yes there are more cases, I can change them as well. I discovered this with clippy::pedantic. I didn't have time to look through all things clippy::pedantic highlighted, that is beyond the scope of this PR and is also a lot of work.

Comment (thomaseizinger) [reply examples/chat-tokio.rs:56]: We ignore `clippy::pedantic` in here (see `custom-clippy` alias in `.cargo/config.toml`)

File: core/CHANGELOG.md (hunk 1 above)
Comment (thomaseizinger) [reply core/CHANGELOG.md]: Agreed, lets not clutter the changelog with such details.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
I am okay to merge this if we drop the changelog entries :)

File: core/CHANGELOG.md (hunk 1 above)
Comment (umgefahren) [reply core/CHANGELOG.md]: I removed the changes from the changelog.

**Review by thomaseizinger — APPROVED** [review]

**Comment by thomaseizinger** [comment]
@mergifyio backport rust-v0480-testplan

**Pull request merged**